smarts-imitation/smarts_imitation/envs/core.py

```diff
@@ -247,23 +295,149 @@ def _init_scenario(self):
                 self.scenario = scenario
                 break
         assert self.scenario is not None
+        self.scenario_name = self.scenario.name
+
+        self._init_vehicle_missions()
 
-        self.vehicle_missions = self.scenario.discover_missions_of_traffic_histories()
-        if self.vehicle_ids is None:
-            self.vehicle_ids = np.array(list(self.vehicle_missions.keys()))
         # TODO(zbzhu): Need further discussion here, i.e., how to maintain BOTH randomness and sorted order of vehicles.
         if self.control_vehicle_num == 1:
             np.random.shuffle(self.vehicle_ids)
             self.vehicle_itr = 0
         else:
             # Sort vehicle id by starting time, so that we can get adjacent vehicles easily.
-            vehicle_start_times = [
-                self.vehicle_missions[v_id].start_time for v_id in self.vehicle_ids
-            ]
-            self.vehicle_ids = self.vehicle_ids[np.argsort(vehicle_start_times)]
+            self.vehicle_ids = self.vehicle_ids[np.argsort(self.vehicle_start_times)]
             self.vehicle_itr = np.random.choice(len(self.vehicle_ids))
+
         self.episode_count = 0
 
     def destroy(self):
         if self.smarts is not None:
             self.smarts.destroy()
+
+    def _vehicle_pos_between(self, vehicle_id: str, start_time: float, end_time: float):
+        """Find the vehicle states between the given history times."""
+        query = """SELECT T.position_x, T.position_y, T.sim_time
+                   FROM Trajectory AS T
+                   WHERE T.vehicle_id = ? AND T.sim_time > ? AND T.sim_time <= ? 
+                   ORDER BY T.sim_time DESC"""
+        rows = self.scenario._traffic_history._query_list(
+            query, (vehicle_id, start_time, end_time)
+        )
+        return (VehiclePosition(*row) for row in rows)
+
+    def _get_vehicle_current_history_position(self, vehicle_id):
```

Comment:
Please add comments to briefly explain the meaning of `current_history_position`.

Reply:
added.